rfcs/centralized_run_criteria.md

```diff
@@ -0,0 +1,87 @@
+# Feature Name: `centralized_run_criteria`
+
+## Summary
+
+This feature lifts run criteria and related machinery out of individual stages, and centralizes them in the schedule instead.
+
+## Motivation
+
+Currently, run criteria are stage-specific: it's not possible to use the same criteria across two stages without creating two copies of it. Code and behavior duplication is bad enough on its own, but here it also results in unintuitive quirks in any mechanism built on top of criteria ([bevy#1671](https://github.com/bevyengine/bevy/issues/1671), [bevy#1700](https://github.com/bevyengine/bevy/issues/1700), [bevy#1865](https://github.com/bevyengine/bevy/issues/1865)).
+
+## User-facing explanation
+
+Run criteria are utility systems that control whether regular systems should run during a given schedule execution. They can be used by entire schedules and stages, or by individual systems. Any system that returns `ShouldRun` can be used as a run criteria.
+
+Criteria can be defined in two ways:
+- Inline, directly attached to their user:
+```rust
+fn player_has_control(controller: Res<PlayerController>) -> ShouldRun {
+    if controller.player_has_control() {
+        ShouldRun::Yes
+    } else {
+        ShouldRun::No
+    }
+}
+
+stage
+    .add_system(player_movement_system.with_run_criteria(run_if_player_has_control))
+    .add_system(
+        player_attack_system.with_run_criteria(|controller: Res<PlayerController>| {
+            if controller.player_can_attack() {
+                ShouldRun::Yes
+            } else {
+                ShouldRun::No
+            }
+        }),
+    );
+```
+- Globally, owned by a schedule, usable by anything inside said schedule via a label:
```

Comment:
This schedule-based API is very useful for raw consumers of `bevy_ecs`, but less familiar for users of `bevy_ecs`, who may not clue in that you can just add this to the `AppBuilder` as well.

Reply:
I want to say that this RFC is about features of `bevy_ecs`, `AppBuilder` may as well not exist.

Reply:
Absolutely, but `bevy_app` *is* a downstream consumer, and will have to respond to these API changes.

Not essential here though, just something to bear in mind.